api/app/signals/apps/api/serializers/attachment.py

```diff
@@ -40,30 +67,39 @@ class Meta:
         extra_kwargs = {'file': {'write_only': True}}
 
     def create(self, validated_data):
-        attachment = Signal.actions.add_attachment(validated_data['file'], self.context['view'].get_signal())
-
-        if self.context['request'].user:
-            attachment.created_by = self.context['request'].user.email
-            attachment.save()
-
-        return attachment
-
-    def validate_file(self, file):
-        if file.size > settings.API_MAX_UPLOAD_SIZE:
-            msg = f'Bestand mag maximaal {settings.API_MAX_UPLOAD_SIZE} bytes groot zijn.'
+        signal = self.context['view'].get_signal()
+        if signal.status.state not in PUBLIC_UPLOAD_ALLOWED_STATES:
```

Comment:
I like this solution better than not allowing any attachment after X time.

However we need to talk about this. In the refinement session of Amsterdam was a story that will conflict with this solution.